openlcsd/flow/tasks.py

```diff
@@ -751,6 +750,40 @@ def get_components_product_from_corgi(context, engine):
     engine.logger.info("Finished getting container components data.")
 
 
+def get_remote_source_components(context, engine):
+    package_nvr = context.get('package_nvr')
+    config = context.get('config')
+    koji_connector = KojiConnector(config)
+    engine.logger.info('Start to get remote source components...')
+    try:
+        binary_nvr = koji_connector.get_binary_nvr(package_nvr)
+        binary_build = koji_connector.get_build(binary_nvr)
+        rs_components = koji_connector.get_remote_source_components(
+            binary_build)
+    except (RuntimeError, ValueError) as err:
+        err_msg = f"Failed to get remote source components. Reason: {err}"
+        engine.logger.error(err_msg)
+        raise RuntimeError(err_msg) from None
+    engine.logger.info('Finished getting remote source components.')
+    return rs_components
+
+
+def get_components_product_from_source_container(context, engine):
+    engine.logger.info('Start to get components data from source container...')
+    config = context.get('config')
+    sc_nvr = context.get('package_nvr')
+    # Get components from the source container itself.
+    srpm_dir = context.get('srpm_dir')
+    misc_dir = context.get('misc_dir')
+    sc_handler = SourceContainerHandler(config)
+    components = sc_handler.get_container_components(srpm_dir, misc_dir,
+                                                     sc_nvr)
+    # Get remote source components from its binary container.
+    components.update(get_remote_source_components(context, engine))
```

Comment:
It's ok for me to update the components. Getting remote source components to the sc_handler.get_container_components is more consistent.

Reply:
They use different functions, srpms use function in sc_handler, but for remote source, get component form koji, so use the function defined in `kojiconnector.py`